sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/RntbdTransportClient.java

```diff
@@ -117,22 +117,30 @@ public long id() {
         URI address = addressUri.getURI();
 
         final RntbdRequestArgs requestArgs = new RntbdRequestArgs(request, address);
+
         requestArgs.traceOperation(logger, null, "invokeStoreAsync");
 
         final RntbdEndpoint endpoint = this.endpointProvider.get(address);
         final RntbdRequestRecord record = endpoint.request(requestArgs);
 
         logger.debug("RntbdTransportClient.invokeStoreAsync({}, {}): {}", address, request, record);
 
-        return Mono.fromFuture(record).doFinally(signalType -> {
-            logger.debug("SignalType.{} received from reactor: {\n  endpoint: {},\n  record: {}\n}",
-                signalType.name(),
-                endpoint,
-                record);
-            if (signalType == SignalType.CANCEL) {
-                record.stage(RntbdRequestRecord.Stage.CANCELLED_BY_CLIENT);
+        return Mono.fromFuture(record.whenComplete((response, throwable) -> {
+
+            record.stage(RntbdRequestRecord.Stage.COMPLETED);
+
+            if (throwable == null) {
+                response.setRequestTimeline(record.takeTimelineSnapshot());
+            } else {
+                checkArgument(throwable instanceof CosmosClientException, "expected %s, not %s: %s",
+                    CosmosClientException.class,
+                    throwable.getClass(),
+                    throwable);
+                CosmosClientException error = (CosmosClientException) throwable;
+                error.setRequestTimeline(record.takeTimelineSnapshot());
```

Comment:
are we not record cancellation triggered by reactor-stream now?

Reply:
what happens on cancellation triggered by reactor-stream are they recorded anywhere?

Reply:
You found a bug, thanks. I found on my local branch that I must capture cancellations from the reactor stream. I'll add that in the morning and update issue #7250 to ensure there's test coverage when that issue is resolved.